classes/module/Module.php

```diff
@@ -982,8 +986,23 @@ public function disable($force_all = false)
 
         // Disable module for all shops
         $sql = 'DELETE FROM `' . _DB_PREFIX_ . 'module_shop` WHERE `id_module` = ' . (int) $this->id . ' ' . ((!$force_all) ? ' AND `id_shop` IN(' . implode(', ', Shop::getContextListShopID()) . ')' : '');
+        $resultDeletionQuery = Db::getInstance()->execute($sql);
+
+        // if module has no more shop associations, set module.active = 0
+        if (!$this->hasShopAssociations()) {
+            $disableQuery = 'UPDATE %smodule SET active = 0 WHERE id_module = %s';
+            $resultDisableQuery = Db::getInstance()->execute(sprintf($disableQuery, _DB_PREFIX_, $this->id));
```

Comment:
```suggestion
            $resultDisableQuery = Db::getInstance()->update('module', ['active' => 0], 'id_module = '. (int) $this->id);
```

With the remove of `$disableQuery`